File: Two_Pointers/125_valid_palindrome/valid_palindrome.cpp

#include <iostream>
#include <cctype>
using namespace std;

bool isPalindrome(string s) {

    int i = 0;
    int j = s.size() - 1;

    while (i < j) {
    
            while ((i < j) && (isalnum(s[i]) == false)) {
                i++;
            }
            
            while ((i < j) && (isalnum(s[j]) == false)) {
                j--;
            }
            
            if (tolower(s[i]) != tolower(s[j])) {
                return false;
            }
            
            i++;
            j--;
        }
        return true; 
}


int main() {

    string s = "A man, a plan, a canal: Panama";

    cout << "result : " << isPalindrome(s) << endl;

    return 0;
}

File: Two_Pointers/125_valid_palindrome/valid_palindrome.py


def isPalindrome(s: str) -> bool:
        
        i = 0
        j = len(s) - 1
        
        while i < j:
            while i < j and s[i].isalnum() == False:
                i += 1
            while i < j and s[j].isalnum() == False:
                j -= 1
                
            if s[i].lower() != s[j].lower():
                return False
            
            i += 1
            j -= 1
        
        return True

s = "A man, a plan, a canal: Panama"

print("result : ", isPalindrome(s))